cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/GoalViolations.java

```diff
@@ -23,73 +23,67 @@
 public class GoalViolations extends KafkaAnomaly {
   private static final Logger LOG = LoggerFactory.getLogger(GoalViolations.class);
   private final KafkaCruiseControl _kafkaCruiseControl;
-  private final List<Violation> _goalViolations = new ArrayList<>();
+  // The priority order of goals is maintained here.
+  private final List<String> _fixableViolatedGoals;
+  private final List<String> _unfixableViolatedGoals;
```

Comment:
Rather than having two independent lists, would it make our implementation more clear to have a single `Map<Boolean, List<String>>` for these two?